### PR TITLE
re #147183. Support editor options in IDefaultEditor

### LINKED ISSUE #147183: Support index based url for notebook cells
Provide a  url for cell locations like `https://github.com/owner/repo/blob/master/notebook.ipynb#C2:L6-10`

### PR DESCRIPTION
**Pull request by rebornix**
This PR fixes #147183. It attempts to introduce `IEditorOptions` to `IDefaultLayout`, which can be transported to any workbench editor. It will allow embedders to construct a notebook editor options for notebook resources.

### CODE REVIEW

File: src/vs/workbench/common/editor.ts [hunk 1]
```diff
@@ -1340,11 +1340,12 @@ export async function pathsToEditors(paths: IPathData[] | undefined, fileService
...
 		}
 
-		const options: ITextEditorOptions = {
+		const options = {
+			...path.options,
 			selection: exists ? path.selection : undefined,
 			pinned: true,
 			override: path.editorOverrideId
-		};
+		} as IEditorOptions;
```
Comment (bpasero) [inline src/vs/workbench/common/editor.ts:1348]: I prefer to declare the type above in line 1343 over explicit cast:

```ts
const options: IEditorOptions = ...
```

File: src/vs/platform/window/common/window.ts [hunk 2]
```diff
@@ -178,6 +179,11 @@ export interface IPathData {
 	// the file path to open within the instance
 	readonly fileUri?: UriComponents;
 
+	/**
+	 * Optional editor options to apply in the file
+	 */
+	readonly options?: IEditorOptions;
```
Comment (bpasero) [inline src/vs/platform/window/common/window.ts:185]: Should also set deprecation to `selection` and `editoOverrideId` here.

**Review by bpasero — CHANGES_REQUESTED** [review]
Only minor feedback, please merge after, no need to review again.

File: src/vs/platform/window/common/window.ts (hunk 2 above)
Comment (rebornix) [reply src/vs/platform/window/common/window.ts:185]: 👍

File: src/vs/workbench/common/editor.ts (hunk 1 above)
Comment (rebornix) [reply src/vs/workbench/common/editor.ts]: Yep, did the casing as object literal doesn't like unknown properties. Tweaked it.

**Pull request merged**